### PR TITLE
Classifier: localise the field guide

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Pass translation `strings` down to field guide components.
- Use the `strings` dictionary to show item titles and content.

## Package
lib-classifier

## How to Review
Load PH-TESS in the dev classifier and change language to change the field guide language.
https://localhost:8080/?project=nora-dot-eisner/planet-hunters-tess&env=production

https://user-images.githubusercontent.com/59547/169847369-95cb305d-23f3-418f-949c-391534e59ce2.mov

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

**Review by goplayoutside3 — COMMENTED** [review]
Can confirm that TESS's field guide is translated when switching locales. Noting here though that the field guide storybook does not load titles or content on this branch.

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideConnector.js [hunk 1]
```diff
@@ -1,50 +1,20 @@
...
-function useStores() {
-  const stores = React.useContext(MobXProviderContext)
-
-  const { 
-    active: fieldGuide,
-    attachedMedia: icons,
-    activeItemIndex,
-    setActiveItemIndex,
-    setModalVisibility,
-    showModal
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideConnector.js:14]: Specifically `showModal`.

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItem/FieldGuideItem.js [hunk 2]
```diff
@@ -42,69 +42,64 @@ const markdownComponents = {
...
+        <StyledButton
+          a11yTitle={t('FieldGuide.FieldGuideItem.ariaTitle')}
+          icon={<FormPrevious color='light-5' />}
+          margin={{ right: 'small' }}
+          onClick={() => setActiveItemIndex()}
+          plain
+        />
+        <Markdownz components={markdownTitleComponent}>
+          {`### ${title}`}
+        </Markdownz>
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItem/FieldGuideItem.js:76]: The `###` Markdown code can be removed too.

**Comment by eatyourgreens** [comment]
I keep forgetting to add translation strings to the storybook.

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideConnector.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/FieldGuide/FieldGuideConnector.js:14]: I think they can all be removed.

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItem/FieldGuideItem.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItem/FieldGuideItem.js:76]: I wonder if this means that there are projects that do use Markdown in the item titles?

**Comment by eatyourgreens** [comment]
It seems to be intermittent. Here's a screen recording where the text vanishes briefly at the end. Leaving the tab and changing back triggers the field guide story to refresh.
https://user-images.githubusercontent.com/59547/171488730-17bf2bf0-1da2-4fa2-8791-c27ec7117af6.mov

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItemAnchor/FieldGuideItemAnchor.js [hunk 3]
```diff
@@ -18,56 +24,50 @@ export function AnchorLabel ({ className, icons, item }) {
     >
       <FieldGuideItemIcon alt={item.title} fit='cover' height='100px' icon={icon} width='100px' />
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/components/FieldGuideItemAnchor/FieldGuideItemAnchor.js:25]:
```suggestion
      <FieldGuideItemIcon alt={title} fit='cover' height='100px' icon={icon} width='100px' />
```

**Review by goplayoutside3 — APPROVED** [review]
Great! Looks like storybook is working now 👍 

Noting here that this PR changes the UX of the field guide a little bit. On master, let's say someone opens PH-TESS's field guide, clicks on Planet Transits, and then closes the field guide modal. The next time they open the field guide it will still display Planet Transits. With this PR, closing the field guide modal will always "reset" it so the next time it's opened the menu of items will display. I don't think it's a big enough change for concern, and I think it makes more sense for the active field guide item logic to be handled in the FieldGuide component as in this PR. But that's my opinionated statement, so just want to document here!

**Comment by eatyourgreens** [comment]
If volunteers want the old behaviour for the field guide, I think it would be easy enough to bring back. 

I find global state in the window can be hard to debug (thinking of some of the bugs that you found in the tutorial.) I totally agree about keeping the field guide item state as close to the item code as possible.

**Pull request merged**